### DISCUSSION TITLE
Mutagen, Exaile and Quodlibet

### DISCUSSION THREAD

**Discussion by mister-bum** [exaile/exaile#916] [Q&A]
Sorry, if this is not the right place for my question, but maybe it is....

I tried out quodlibet a couple of months ago, and came across a rather strange behavior/error that is related to the database and the tags.
You can find the corresponding issue here: https://github.com/quodlibet/quodlibet/issues/3869

In search for an alternative I came across exaile and read that it uses the mutagen module too. I do not think that the quodlibet error is mutagen related, but I cannot be sure of course. 

So this post is merely to double-check whether somebody from the exaile-community can remember a similar error/behavior/issue somewhere along the development path of exaile. 

Unfortunately my coding skills are not solid enough to dig into the quodlibet sources, but maybe I can do some (hopefully useful) research this way

**Comment by sjohannes** [comment]
My understanding from reading the ticket is that TPE2 is officially the Performer tag, but people have been using it for Album Artist; and Quod Libet have been hesitant to change this due to questions about backwards compatibility.

Exaile also used to map TPE2 to Performer, but we [changed](https://github.com/exaile/exaile/commit/77177a681f1d830a40b20cb9edc396eea0837d68) it to Album Artist around 13 years ago. I don't remember if there was any discussion around this, and I don't think we did anything to fix the resulting inconsistency in users' files.

We did recently have to change the meaning of some other tags, so I understand the [complexity](https://github.com/exaile/exaile/pull/871/files#diff-20d9cb13f0efc616fad4512e7d6ac335c0581aeafd9934eaa0ffb610bb6aec6eR170-R171) involved in trying to do this while attempting to minimise disruptions.

Edit: The discussion for our recent tag changes is in <https://github.com/exaile/exaile/issues/682>. I think we ended up showing a message box on first startup after upgrade, and a button to do the migration on files that are in the collection.
